src/dataServices/dynamoDbDataService.ts

```diff
@@ -97,13 +99,16 @@ export class DynamoDbDataService implements Persistence, BulkDataAccess {
 
     async createResource(request: CreateResourceRequest) {
         const { resourceType, resource } = request;
+        return this.createResourceWithId(resourceType, resource, uuidv4());
+    }
 
+    private async createResourceWithId(resourceType: string, resource: any, resourceId: string) {

```

Comment:
Could we add a regex validation for resourceId? 

I believe this regex will do the job: `^[\w\-]{1,64}$`

[FHIR spec](https://www.hl7.org/fhir/resource.html#id): "Ids can be up to 64 characters long, and contain any combination of upper and lowercase ASCII letters, numerals, "-" and ".""

Reply:
Yes good idea.  
We could end up with a case where resources that are `created` have a different id format to those that are `updateCreated`, do we care about this?

Reply:
From a technical stand-point I don't see any issues with a Dynamo key (id) being different or ES param (id) being different.

One issue that may come up is create a 'hot' shard meaning if all the keys started to look the same that shard of Dynamo may suffer in performance. I don't think a regex would solve this and I think anything but random can. So with us allowing `updateCreate` there will be some additional risk if users are malicious in crafting their ids

Reply:
Agree.  An application could use solely `updateCreate` operations to create resources with a different style of id.  If these are not random and performance is affected then that is down to poor application design rather than a server implementation issue.